rfcs/rfc-TestRails.md

```diff
@@ -0,0 +1,46 @@
+
+## Motivation
+
+Currently BC ROS COLIN project does not have a Test Case Management Tool. This lack of tool has created a few headaches in our current testing process. The following is a brief overview of them.
+
+- Difficult to document test plan in a way that is easy to understand
+- It is time intensive to offload testing onto someone else, be they a part of the QA team or not. Since without a test plan, and familiarity of the product, they are unsure of what is a bug or a feature
+- Tracking test execution and bug reporting is currently barely possible in google sheets .
+- There is no test execution report at the end of the project
+- Increase the time needed for someone to automate testing, as the test case must be written before coded, and there is no current written test cases
+- An increased risk of bugs in productions due to inconsistent testing procedures
+
+## Recommendation
+
+- Purchase 5 TestRail licences to be shared by the QA team.
+- Cost is $32 USD/user/month * 5 = $160/month
+- TestRail Cloud licences chosen over on-prem
+
+Having a test case management tool would have many benefits, but the two main benefits are in saving time and improving reliability. There are three main places that time will be saved and one where reliability will be increased.
+
+- Firstly, a large amount of time will be saved in the writing of the test cases.
+- Secondly, time will be saved when executing test cases as the steps and results are explicit and detailed. This will also improve the reliability of the software, as testing will be consistent and regression issues will be quickly found.
+- Thirdly, time would be saved when trying to export test cases and share testing results to clients.
+
+## Alternatives
+
+A test case management tool would greatly solve all the issues mentioned above. Over the last few weeks a  few different tools were researched with the following results
+
+Tool | Pro | Con | Pricing (USD)
+---- | --- | --- | ---
+Kuality | Free to use under 5 user, Can export tests in Word, CSV, Excel  | Cloud App, Difficult to navigate, Requires a lot of set up, Reports are not easy to read  | $45/user/month
+Test Rails | Easy Github and zenhub Integration, Postman and other automation tools, Easy import and export of test cases, Easy to get started, Bulk editing cases and adding results | Cloud Solution, Reports aren't easy to export | $32/user/ month
+Zephyr | Already integrated in JIRA, All JIRA users have access, Easy to create reports | Cost per JIRA user, BC ROS is not using JIRA and Zephyr is a JIRA plugin | $20/user/month
+
+## Adoption strategy
+
+QA must work with the planning/business to document all test cases from at least the feature level. Value is immediately lost if this suites get out of date. More work is needed to firm up when the scripts will be drafted and which resources will be primarily engaging in this work. Manual test runs will be tracked initially with ongoing work to determine:
```

Comment:
In reference to "The value is immediately lost if the suites get out of date":  

You're absolutely right - maintaining these tests cases are just as essential as maintaining the unit, integration, regression and e2e tests. To help ensure that this doesn't get missed, it should be thought of in sprint-planning, backlog grooming.  When defining a feature to be developed, updating test cases to include any changes or additions brought about from this feature should be included as part of that features development (This is typically done by a QA but I've seen it done by BAs and Devs too ).  Potentially updating the DoD/DoR to reference this ?

Reply:
In general I strongly feel this is a good path to take. There is a lot of value that can be gained from implementing this as a Test Case Management tool. I definitely vote yes to picking this up and starting to use it.